cmd/snap-fde-keymgr/main.go

```diff
@@ -0,0 +1,123 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"github.com/jessevdk/go-flags"
+
+	"github.com/snapcore/snapd/secboot"
+	"github.com/snapcore/snapd/secboot/keymgr"
+)
+
+var osStdin io.Reader = os.Stdin
+
+type commonDeviceMixin struct {
+	Device string `long:"device" description:"encrypted device" required:"yes"`
+}
+
+type cmdAddRecoveryKey struct {
+	commonDeviceMixin
+	KeyFile string `long:"key-file" description:"path to recovery key file" required:"yes"`
+}
+
+type cmdRemoveRecoveryKey struct {
+	commonDeviceMixin
+	KeyFile string `long:"key-file" description:"path to recovery key file" required:"yes"`
+}
+
+type cmdChangeEncryptionKey struct {
+	commonDeviceMixin
+}
+
+type options struct {
+	CmdAddRecoveryKey      cmdAddRecoveryKey      `command:"add-recovery-key"`
+	CmdRemoveRecoveryKey   cmdRemoveRecoveryKey   `command:"remove-recovery-key"`
```

Comment:
we might want to make these two commands operate on more than one disk at a time?

Reply:
I'll add a TODO about that